### PR TITLE
feature: basic image setting working

### PR DESCRIPTION
**Pull request by recognitionone**

### CODE REVIEW

File: src/controller.js [hunk 1]
```diff
@@ -1,20 +1,53 @@
...
 export class MyCounterController {
 	constructor() {
 		this.view = new MyCounterView();
-		this.model = new MyCounterModel(10, 0);
+		this.model = new MyCounterModel(15, 0);
+
+		this.imagesModel = new ImagesModel(imagesMock);
+
+
+		// this.obj = this.imagesModel.myObj;
```
Comment (mika1111) [inline src/controller.js]: Remove commented code

File: src/controller.js [hunk 2]
```diff
@@ -1,20 +1,53 @@
...
 	}
 
 	init() {
-		this.view.addStartHandler(() => { this.model.start() });
+		this.view.addStartHandler(() => { this.model.start(), this.setImage(this.imagesModel.timeLaps[0]) });
 		this.view.addPauseHandler (() => { this.model.pause() });
-		this.view.addResetHandler (() => { this.model.reset() });
+		this.view.addResetHandler (() => { this.model.reset(), this.view.setupText("reset"), this.setImage( this.imagesModel.failedButonimage ) });
```
Comment (mika1111) [inline src/controller.js]: Split to several lines or move to some function

File: src/controller.js [hunk 3]
```diff
@@ -1,20 +1,53 @@
...
+		this.setImage(this.imagesModel.beforeButonimage);
+
+		this.model.counterEvent.addEventListener("changeValue", 
+			(e) => {
+					console.log(this.imagesModel.addImage(e.detail.text));
```
Comment (mika1111) [inline src/controller.js]: Remove console log

File: src/controller.js [hunk 4]
```diff
@@ -1,20 +1,53 @@
...
+					
+			} )
+
+
+
+
+
+
+		this.addListener("koniec", this.imagesModel.successButonimage);
+		// this.addListener("step1",  this.obj.timeLaps[0]);
```
Comment (mika1111) [inline src/controller.js]: Remove comments

File: src/controller.js [hunk 5]
```diff
@@ -1,20 +1,53 @@
...
+		// this.addListener("step4",  this.obj.timeLaps[15]);
+	}
+
+	setImage(name) {
+		return this.view.setupImage( name );		
+	}
 
-		this.model.counterEvent.addEventListener("reset", () => {this.view.setupText("reset")} );
-		this.model.counterEvent.addEventListener("changeValue", (e) => {this.view.setupText(e.detail.text)} )
+	addListener(eventName, imageLink) {
```
Comment (mika1111) [inline src/controller.js]: Change name for addImageChangeListener

File: src/imagesModel.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+
+
+export class ImagesModel {
+	constructor(imagesData) {
+		//TODO - sprawdzanie/ validacja danych 
+		Object.assign(this, imagesData);
+	}	
+
+
+	addImage(timer) {
```
Comment (mika1111) [inline src/imagesModel.js]: change to findImage or getImage

File: src/model.js [hunk 7]
```diff
@@ -8,18 +8,52 @@ export class MyCounterModel {
...
-				this.counterEvent.dispatchEvent(new Event("koniec"));
+				// this.counterEvent.dispatchEvent(new Event("koniec"));
+				this.createCustomEvent("koniec")
+
 			} else {
 				this.currentValue--;
-				this.counterEvent.dispatchEvent(new CustomEvent("changeValue", { detail: { text: this.currentValue} }));
+				// this.counterEvent.dispatchEvent(new CustomEvent("changeValue", { detail: { text: this.currentValue} }));
+				this.createCustomEvent("changeValue", { detail: { text: this.currentValue} });
+				switch (this.currentValue) {
```
Comment (mika1111) [inline src/model.js]: Remove switch

File: src/model.js [hunk 8]
```diff
@@ -8,18 +8,52 @@ export class MyCounterModel {
...
 	start() {
-		this.counterEvent.dispatchEvent(new Event("start"));
+		// this.counterEvent.dispatchEvent(new Event("start"));
+		this.createCustomEvent("start")
+
 		this.timer = setInterval(() => {
 			if(this.currentValue === this.stopValue) {
 				clearInterval(this.timer);
-				this.counterEvent.dispatchEvent(new Event("koniec"));
+				// this.counterEvent.dispatchEvent(new Event("koniec"));
```
Comment (mika1111) [inline src/model.js]: Remove comments

File: src/model.js [hunk 9]
```diff
@@ -8,18 +8,52 @@ export class MyCounterModel {
...
+				  	break;
+				  default:
+				}
+
+
 			}
 		}, 1000);
 	}
 
+	createCustomEvent(eventName, eventDetail) {
```
Comment (mika1111) [inline src/model.js]: Great :)

Comment (recognitionone) [reply src/model.js]: :)

File: mock/imagesSchema.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (mika1111) [inline mock/imagesSchema.js]: We could try avj on separate branch, remove it from this branch

File: package.json [hunk 11]
```diff
@@ -27,5 +27,7 @@
     "webpack": "^4.31.0",
     "webpack-cli": "^3.3.2"
   },
-  "dependencies": {}
+  "dependencies": {
+    "ajv": "^6.10.0"
```
[the remark this answers is not in the document]
Comment (recognitionone) [reply package.json]: ok

File: mock/imagesSchema.js (hunk 10 above)
Comment (recognitionone) [reply mock/imagesSchema.js]: ok

**Pull request merged**